src/test/kotlin/com/terraformation/backend/accelerator/AcceleratorSearchTest.kt

```diff
@@ -112,4 +114,23 @@ class AcceleratorSearchTest : DatabaseTest(), RunsAsUser {
 
     assertJsonEquals(expected, searchService.search(prefix, fields, NoConditionNode()))
   }
+
+  @Test
+  fun `can filter organizations by internal tag to only retrieve accelerator data`() {
+    val nonAcceleratorOrgId = insertOrganization(2)
+
+    insertOrganizationUser(organizationId = organizationId, role = Role.TerraformationContact)
+    insertOrganizationUser(organizationId = nonAcceleratorOrgId, role = Role.Admin)
+
+    every { user.organizationRoles } returns
+        mapOf(organizationId to Role.TerraformationContact, nonAcceleratorOrgId to Role.Admin)
+
+    val prefix = SearchFieldPrefix(searchTables.organizations)
+    val fields = listOf(prefix.resolve("name"))
+    val condition = FieldNode(prefix.resolve("internalTags_name"), listOf("Accelerator"))
+
+    val expected = SearchResults(listOf(mapOf("name" to "Organization 1")), null)
+
+    assertJsonEquals(expected, searchService.search(prefix, fields, condition))
+  }
 }
```

Comment:
I think as we expand search functions. It is important for us to test exceptions on not-permissible cases.